pkg/api/v1/types.go

```diff
@@ -100,11 +100,14 @@ func init() {
 	})
 }
 
+// VM is *the* VM Definition. It represents a virtual machine in the runtime environment of kubernetes.
 type VM struct {
 	metav1.TypeMeta `json:",inline"`
 	ObjectMeta      metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec            VMSpec            `json:"spec,omitempty" valid:"required"`
-	Status          VMStatus          `json:"status"`
+	// Spec contains the low level VM specification that can be understood by underlying libraries.
```

Comment:
Hm, in Kubernetes we don't really have the mentioned distinction here. All the v1 objects (e.g. v1.Pod from k8s, but also v1.VM from us), are used for inter-service communication, but also serve as user-facing API (makes sense, when you think about controllers as operators/users).

There exist "internal" representations in k8s though. For instance a v1.Pod has a corresponding api.Pod, where api.Pod is the etcd representation, but that never leaves the apiserver. In the KubeVirt context, right now we don't have this DAO representation yet (but will come with the aggregated api server), but we have something similar in virt-handler, where we have Domains derived from VMs, and these Domain structs also never leave the service.